internal/services/watchlist/watchlist.go

```diff
@@ -0,0 +1,83 @@
+package watchlist
+
+import (
+	"Tradeasy/config"
+	"Tradeasy/internal/model"
+	"errors"
+)
+
+func CreateWatchlist(Req CreateRequest) (Res CreateResponse, err error) {
+	var user model.Users
+	err = config.DB.Table("users").Where("user_id = ?", Req.UserId).First(&user).Error
+	if err != nil {
+		return Res, errors.New("user not found")
+	}
+	var wl model.Watchlist
+	err = config.DB.Table("watchlist").Where("name = ? AND user_id = ?", Req.WatchlistName, Req.UserId).First(&wl).Error
+	if err == nil {
+		return Res, errors.New("watchlist name already exists")
+	}
+	wl.Name = Req.WatchlistName
+	err = config.DB.Table("watchlist").Create(&wl).Error
+	if err != nil {
+		return Res, errors.New("failed to create watchlist")
+	}
+	var uwl model.UserWatchlist
+	uwl.WatchlistId = wl.Id
+	uwl.Userid = Req.UserId
+	err = config.DB.Table("user_watchlist").Create(&uwl).Error
+	if err != nil {
+		return Res, errors.New("failed to create watchlist")
+	}
+	Res.Message = "Watchlist created"
+	Res.WatchlistId = wl.Id
+
+	return Res, nil
+}
+
+func AddStockEntry(Req AddStockRequest, watchlistId int) (Res AddStockResponse, err error) {
+	var uwl model.UserWatchlist
+	err = config.DB.Table("user_watchlist").Where("user_id = ? AND watchlist_id = ?", Req.UserId, watchlistId).First(&uwl).Error
+	if err != nil {
+		return Res, errors.New("user not found")
+	}
+
+	err = config.DB.Table("user_watchlist").Where("stock_name = ? AND watchlist_id = ?", Req.StockName, watchlistId).First(&uwl).Error
+	if err == nil {
+		return Res, errors.New("stock name already exists")
+	}
+
+	uwl.StockName = Req.StockName
+	config.DB.Table("user_watchlist").Where("user_id = ? AND watchlist_id = ?", Req.UserId, watchlistId).Update("stock_name", Req.StockName)
```

Comment:
why update?

Reply:
coz watchlist_id and user_id are already there in the user_watchlist table